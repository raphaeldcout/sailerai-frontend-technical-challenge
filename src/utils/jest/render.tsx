import { FC, ReactElement, ReactNode } from 'react';

import { ThemeProviderSailer } from '@/styles/withThemeProvider';
import { render, RenderOptions } from '@testing-library/react';

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <ThemeProviderSailer>{children}</ThemeProviderSailer>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };

