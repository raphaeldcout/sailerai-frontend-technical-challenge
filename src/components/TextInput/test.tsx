import '@testing-library/jest-dom';

import { render } from '@/utils/jest/render';

import { TextInput } from '.';

describe('TextInput', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <TextInput id="text-input" label="Label" value="" onChange={jest.fn()} />
    );

    expect(container).toMatchSnapshot();
  });
});
