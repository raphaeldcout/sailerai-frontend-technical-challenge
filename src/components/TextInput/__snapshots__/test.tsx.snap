// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should match snapshot 1`] = `
<div>
  <div
    class="sc-dc7a9aa4-0 iNLJyx"
  >
    <input
      class="sc-dc7a9aa4-2 iomJIg"
      id="text-input"
      placeholder=" "
      value=""
    />
    <label
      class="sc-dc7a9aa4-1 koBPAR"
      for="text-input"
    >
      Label
    </label>
  </div>
</div>
`;
